### PR TITLE
fix: circular deps check in CI

### PR DESCRIPTION
**Pull request by IvayloAndonov**
# Context
Currently `circular-deps:check` script does not exit appropriately if there are circular refs in the compiled source code. This is cause by `--summary` flag we use for more structured and beautiful output but it masks the exiting and always exits with `0` which is not documented.

# Important Changes Introduced
- remove `--summary` flag so exit codes are correct
- resolve all circular refs within packages

# Note 
Packages `projection-typeorm` and `core` have circular dependencies logically which is fine. 
For example, in `typeorm` 2 distinct entities are referencing each other from two files.
In `core` the case is with `Address` and `BaseAddress` ts files references.

### CODE REVIEW

File: packages/cardano-services/src/Asset/AssetHttpService.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import * as OpenApiValidator from 'express-openapi-validator';
 import { AssetProvider } from '@cardano-sdk/core';
 import { HttpService } from '../Http';
 import { Logger } from 'ts-log';
-import { ServiceNames } from '../Program';
+import { ServiceNames } from '../Program/programs/types';
```
Comment (mkazlauskas) [inline packages/cardano-services/src/Asset/AssetHttpService.ts:5]: This appears to be in a wrong file as it's not a type. Not required to resolve to merge this PR

**Review by mkazlauskas — APPROVED** [review]
Nice work!

**Review by iccicci — APPROVED** [review]
Very nice!

**Review by iccicci — CHANGES_REQUESTED** [review]
Better to change `madge --circular dist` to `madge --circular dist/cjs 2> /dev/null`.
`dist` to `dist/cjs` to make it faster;
`2> /dev/null` to produce a more clean output.
This in all `package.json` files

**Review by iccicci — APPROVED** [review]

**Comment by IvayloAndonov** [comment]
Squashed and ready for merge @mkazlauskas  @iccicci

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**